sky/server/requests/requests.py

```diff
@@ -453,75 +519,98 @@ def kill_requests(request_ids: Optional[List[str]] = None,
+                # request.
+                try:
+                    os.kill(request_record.pid, signal.SIGTERM)
+                except ProcessLookupError:
+                    logger.debug(f'Process {request_record.pid} not found')
             request_record.status = RequestStatus.CANCELLED
             request_record.finished_at = time.time()
             cancelled_request_ids.append(request_id)
     return cancelled_request_ids
 
 
-def create_table(cursor, conn):
+def create_table():
     # Enable WAL mode to avoid locking issues.
     # See: issue #1441 and PR #1509
     # https://github.com/microsoft/WSL/issues/2395
     # TODO(romilb): We do not enable WAL for WSL because of known issue in WSL.
     #  This may cause the database locked problem from WSL issue #1441.
-    if not common_utils.is_wsl():
+    if (_SQLALCHEMY_ENGINE.dialect.name
+            == db_utils.SQLAlchemyDialect.SQLITE.value and
+            not common_utils.is_wsl()):
         try:
-            cursor.execute('PRAGMA journal_mode=WAL')
-        except sqlite3.OperationalError as e:
+            with orm.Session(_SQLALCHEMY_ENGINE) as session:
+                session.execute(sqlalchemy.text('PRAGMA journal_mode=WAL'))
+                session.commit()
+        except sqlalchemy.exc.OperationalError as e:
             if 'database is locked' not in str(e):
                 raise
             # If the database is locked, it is OK to continue, as the WAL mode
             # is not critical and is likely to be enabled by other processes.
 
-    # Table for Requests
-    cursor.execute(f"""\
-        CREATE TABLE IF NOT EXISTS {REQUEST_TABLE} (
-        request_id TEXT PRIMARY KEY,
-        name TEXT,
-        entrypoint TEXT,
-        request_body TEXT,
-        status TEXT,
-        created_at REAL,
-        return_value TEXT,
-        error BLOB,
-        pid INTEGER,
-        {COL_CLUSTER_NAME} TEXT,
-        schedule_type TEXT,
-        {COL_USER_ID} TEXT,
-        {COL_STATUS_MSG} TEXT,
-        {COL_SHOULD_RETRY} INTEGER,
-        {COL_FINISHED_AT} REAL
-        )""")
-
-    db_utils.add_column_to_table(cursor, conn, REQUEST_TABLE, COL_STATUS_MSG,
-                                 'TEXT')
-    db_utils.add_column_to_table(cursor, conn, REQUEST_TABLE, COL_SHOULD_RETRY,
-                                 'INTEGER')
-    db_utils.add_column_to_table(cursor, conn, REQUEST_TABLE, COL_FINISHED_AT,
-                                 'REAL')
-
-
-_DB = None
-_init_db_lock = threading.Lock()
-
-
-def init_db(func):
-    """Initialize the database."""
-
-    @functools.wraps(func)
-    def wrapper(*args, **kwargs):
-        global _DB
-        if _DB is not None:
-            return func(*args, **kwargs)
-        with _init_db_lock:
-            if _DB is None:
+    db_utils.add_tables_to_db_sqlalchemy(Base.metadata, _SQLALCHEMY_ENGINE)
+
+    # For backward compatibility.
+    with orm.Session(_SQLALCHEMY_ENGINE) as session:
```

Comment:
Currently, requests DB is not persisted across API server restarts. It's basically ephemeral. So I'm a little concerned this chance could have confusing implications.
Why is it necessary to put this into postgres instead of having a separate sqlite DB for each replica? Is it mostly to do with `sky api logs/cancel/status`?